core/src/main/java/org/elasticsearch/index/query/HasChildQueryBuilder.java

```diff
@@ -151,12 +153,15 @@ public InnerHitBuilder innerHit() {
     }
 
     public HasChildQueryBuilder innerHit(InnerHitBuilder innerHit) {
-        innerHit.setParentChildType(type);
-        innerHit.setQuery(query);
+        innerHit.setParentChildType(type, query);
         this.innerHitBuilder = innerHit;
         return this;
     }
 
+    public void clearInnerHit() {
+        this.innerHitBuilder = null;
+    }
```

Comment:
Not sure we need this. For instance we don't allow to remove sorts on a SearchRequestBuilder.